crates/accelerate/src/commutation_checker.rs

```diff
@@ -0,0 +1,781 @@
+    "ecr", "ccx", "cswap",
+];
+
+#[pyclass(module = "qiskit._accelerate.commutation_checker")]
+struct CommutationChecker {
+    library: CommutationLibrary,
+    cache_max_entries: usize,
+    cache: HashMap<(String, String), CommutationCacheEntry>,
+    #[pyo3(get)]
+    current_cache_entries: usize,
+    #[pyo3(get)]
+    _cache_miss: usize,
+    #[pyo3(get)]
+    _cache_hit: usize,
+    #[pyo3(get)]
+    gates: Option<HashSet<String>>,
+}
+
+#[pymethods]
+impl CommutationChecker {
+    #[pyo3(signature = (standard_gate_commutations=None, cache_max_entries=1_000_000, gates=None))]
+    #[new]
+    fn py_new(
+        py: Python,
+        standard_gate_commutations: Option<Py<PyAny>>,
+        cache_max_entries: usize,
+        gates: Option<HashSet<String>>,
+    ) -> Self {
+        CommutationChecker {
+            library: CommutationLibrary::new(py, standard_gate_commutations),
+            cache: HashMap::with_capacity(cache_max_entries),
+            cache_max_entries,
+            current_cache_entries: 0,
+            _cache_miss: 0,
+            _cache_hit: 0,
+            gates,
+        }
+    }
+
+    #[pyo3(signature=(op1, op2, max_num_qubits=3))]
+    #[allow(clippy::too_many_arguments)]
+    fn commute_nodes(
+        &mut self,
+        py: Python,
+        op1: &DAGOpNode,
+        op2: &DAGOpNode,
+        max_num_qubits: u32,
+    ) -> PyResult<bool> {
+        let mut bq: BitData<Qubit> = BitData::new(py, "qubits".to_string());
+        op1.instruction
+            .qubits
+            .bind(py)
+            .iter()
+            .for_each(|q| bq.add(py, &q, false).unwrap());
+        op2.instruction
+            .qubits
+            .bind(py)
+            .iter()
+            .for_each(|q| bq.add(py, &q, false).unwrap());
+        let qargs1 = op1
+            .instruction
+            .qubits
+            .bind(py)
+            .iter()
+            .map(|q| bq.find(&q).unwrap().0 as usize)
+            .collect::<Vec<_>>();
+        let qargs2 = op2
+            .instruction
+            .qubits
+            .bind(py)
+            .iter()
+            .map(|q| bq.find(&q).unwrap().0 as usize)
+            .collect::<Vec<_>>();
+
+        let mut bc: BitData<Clbit> = BitData::new(py, "clbits".to_string());
+        op1.instruction
+            .clbits
+            .bind(py)
+            .iter()
+            .for_each(|c| bc.add(py, &c, false).unwrap());
+        op2.instruction
+            .clbits
+            .bind(py)
+            .iter()
+            .for_each(|c| bc.add(py, &c, false).unwrap());
+        let cargs1 = op1
+            .instruction
+            .clbits
+            .bind(py)
+            .iter()
+            .map(|c| bc.find(&c).unwrap().0 as usize)
+            .collect::<Vec<_>>();
+        let cargs2 = op2
+            .instruction
+            .clbits
+            .bind(py)
+            .iter()
+            .map(|c| bc.find(&c).unwrap().0 as usize)
+            .collect::<Vec<_>>();
```

Comment:
Same thing here, you could store the calls to `.bind(py)` in two variables as done in my example above.